var/spack/repos/builtin/packages/readline/package.py

```diff
@@ -45,4 +45,6 @@ class Readline(AutotoolsPackage):
     patch('readline-6.3-upstream_fixes-1.patch', when='@6.3')
 
     def build(self, spec, prefix):
-        make('SHLIB_LIBS=-lncursesw')
+        options = ['SHLIB_LIBS=-L%s -lncursesw', spec['ncurses'].prefix.lib]
```

Comment:
default implementation of `ld_flags` would probably not work as the lib name is `libncursesw` (note the `w`), but you certainly right that the code should at least read
```
options = ['SHLIB_LIBS=-L%s -lncursesw' % spec['ncurses'].prefix.lib]
```